docs: add custom queries example to custom render docs

kierenhughes opened the pull request:
Following on from my previous PR (#151) and having noticed the discussion in #80:

* Adding an example for adding custom queries globally via creating a custom render function.

In the example, I've create a data attribute selector for `data-cy` as this is one of the attributes recommended by Cypress. I think in reality I would favour replacing `data-cy` globally throughout my projects with `data-testid`, but this may be useful for some people not willing to refactor out `data-cy`, or similar, at the moment.

alexkrolick commented on docs/react-testing-library/setup.md:
```diff
@@ -144,6 +144,35 @@ module.exports = {
+import { render, queries, queryHelpers } from '@testing-library/react'
+
+const customQueries = {
+  getByDataCy: queryHelpers.queryByAttribute.bind(
```
It would be nice to show how to use the `buildQueries` function to create a full set of custom queries that conforms to the `get`/`query`/`find` convention

https://github.com/testing-library/dom-testing-library/blob/master/src/query-helpers.js#L103

kierenhughes replied:
Great suggestion - I'll look into it.

kierenhughes replied:
@alexkrolick do you reckon using `buildQueries` in this example specifically would be useful, or shall I get an example added to the 'Helpers' docs for `dom-testing-library`?

alexkrolick replied:
The function should be documented in the API section and it can point to a usage example here

kierenhughes replied:
@alexkrolick added example usage, along with a brief description in the API section

kierenhughes replied:
@alexkrolick any further comments following my updates?

alexkrolick commented on docs/react-testing-library/setup.md:
```diff
@@ -144,6 +144,35 @@ module.exports = {
 </details>
 
+<details>
+<summary>Modify queries globally with custom render</summary>
```
```suggestion
<summary>Customize queries globally with custom render</summary>
```

kierenhughes replied:
Will update! Even if my British instincts make me want to correct it to 'Customi**s**e' 😂

afontcu commented on docs/react-testing-library/setup.md:
````diff
@@ -144,6 +144,35 @@ module.exports = {
+// override render method
+export { customRender as render }
+```
+</details>
````
Looks great! It'd be amazing if, after the config example, we provided a demo usage (as simple as querying a single `<div>` or whatever).
Some people might have a hard time understanding the `queryHelpers`/`buildQueries` API, so a final example showing how to use the newly availalbe `getByDataCy` would come in handy. What do you think about that?

kierenhughes replied:
Fair point! Added a `getByDataCy` usage example 👍

alexkrolick commented:
Should probably explain a little what "cy" stands for but other than that LGTM

kierenhughes commented:
@alexkrolick 

> Should probably explain a little what "cy" stands for but other than that LGTM

I've added a bit about that along with the link to the relevant Cypress.io docs regarding selecting by data-*.

I've also added a recommendation around not needing to use your own queries and only doing so if your queries encourage testing in a user-centric way without focusing on implementation details.

alexkrolick commented:
@all-contributors please add @kierenhughes for docs
